pupil_src/shared_modules/video_capture/utils.py

```diff
@@ -174,34 +174,33 @@ def __init__(self, path: str) -> None:
         self.path = path
         self.ts = None
         self._pts = None
-        self._is_valid = self.check_valid()
+        self._is_valid = None  # calculated on demand
 
     @property
     def is_valid(self):
+        if self._is_valid is None:
+            self._is_valid = self.load_valid_container() is not None
         return self._is_valid
 
-    def check_valid(self):
+    def load_valid_container(self):
         try:
-            cont = av.open(self.path)
+            cont = self.attempt_open_container()
             # Three failure scenarios:
             # 1. Broken video -> AVError
             # 2. decode() does not yield anything
             # 3. decode() yields None
             first_frame = next(cont.decode(video=0), None)
             if first_frame is None:
-                return False  # container does not contain frames
+                return None  # container does not contain frames
         except av.AVError:
-            return False
+            return None
         else:
             cont.seek(0)
-            self.cont = cont
-            return True
+            return cont
 
-    def load_valid_container(self):
-        if self.is_valid:
-            return self.cont
-        else:
-            return None
+    def attempt_open_container(self):
```

Comment:
Since this can raise an `AVError`, the naming is a bit weird. You have to "try attempt open" to use this.
Why not also raise an `InvalidContainerError` (see comment above) and rename this to `open_container()`. Then you can:
```python
try:
  vid.open_container()
except InvalidContainerError:
  # ...
```
Also, since you already hide away the `av.open()` in a separate function (good call btw), this would also hide the `AVError` with a more appropriately-named error.

Reply:
Agreed